ios/brave-ios/Sources/AIChat/Components/Markdown/AttributedString+Markdown.swift

```diff
@@ -0,0 +1,233 @@
+            switch level {
+            case 1:
+              result[range].font = .system(.title).bold()
+            case 2:
+              result[range].font = .system(.title2).bold()
+            case 3:
+              result[range].font = .system(.title3).bold()
+            case 4:
+              result[range].font = .system(.headline).bold()
+            case 5:
+              result[range].font = .system(.body).bold()
+            case 6:
+              result[range].font = .system(.callout).bold()
+            default:
+              break
+            }
+          case .orderedList:
+            listId = intent.identity
+            listType = .ordered
+          case .unorderedList:
+            listId = intent.identity
+            listType = .unordered
+          case .listItem(let index):
+            if listType != .unordered {
+              listType = .ordered
+            }
+            
+            if listType == .ordered {
+              result.characters.insert(contentsOf: "\(index).\t", at: range.lowerBound)
+              result[range].paragraphStyle = previousListId == listId ? AttributedString.listParagraphStyle : AttributedString.lastListParagraphStyle
+            } else {
+              result.characters.insert(contentsOf: "•\t", at: range.lowerBound)
+              result[range].paragraphStyle = previousListId == listId ? AttributedString.listParagraphStyle : AttributedString.lastListParagraphStyle
+            }
+          case .codeBlock(_):
+            result[range].font = preferredFont.monospaced() //.italic()
+            result[range].foregroundColor = AttributedString.codeBlockForegroundColor
+            result[range].paragraphStyle = AttributedString.codeBlockParagraphStyle
+          case .blockQuote:
+            result[range].font = preferredFont.monospaced()
+
+          case .thematicBreak:
+            break
+
+          case .table(_):
+            break
+
+          case .tableHeaderRow:
+            break
+
+          case .tableRow(_):
+            break
+
+          case .tableCell(_):
+            break
+            
+          @unknown default:
+            break
+          }
+        }
+        
+        previousListId = listId
+        
+        if range.lowerBound != result.startIndex {
+          result.characters.insert(contentsOf: "\n", at: range.lowerBound)
+        }
+      }
+    
+    self = result
+  }
+  
+  func trimmingCharacters(in characterSet: CharacterSet) -> AttributedString {
+    let invertedSet = characterSet.inverted
+    
+    let startRange = self.characters.firstIndex(where: {
+      $0.unicodeScalars.allSatisfy(invertedSet.contains(_:))
+    })
+    
+    let endRange = self.characters.lastIndex(where: {
+      $0.unicodeScalars.allSatisfy(invertedSet.contains(_:))
+    })
+
+    guard let startLocation = startRange, let endLocation = endRange else {
+      return self
+    }
+
+    return AttributedString(self[startLocation...endLocation])
```

Comment:
May want to verify that `startLocation` is actually before `endLocation` otherwise this will crash